tests/test_ingestion.py

```diff
@@ -0,0 +1,183 @@
+        cls.calibDataId = {'visit': 229388, 'ccdnum': 1, 'filter': 'z', 'date': '2013-09-01'}
+        cls.defectDataId = {'path': os.path.join('defects', 'D_n20150105t0115_c23_r2134p01_bpm.fits')}
+
+    def setUp(self):
+        self._repo = tempfile.mkdtemp()
+        self._calibRepo = os.path.join(self._repo, 'calibs')
+        templateRepo = os.path.join(IngestionTestSuite.testApVerifyData, 'repoTemplate')
+
+        # Initialize as a valid repository
+        # Can't call copytree on (templateRepo, self._repo) because latter already exists
+        testFiles = os.listdir(templateRepo)
+        for testFile in testFiles:
+            original = os.path.join(templateRepo, testFile)
+            copy = os.path.join(self._repo, testFile)
+            if os.path.isdir(original):
+                shutil.copytree(original, copy)
+            else:
+                shutil.copy2(original, copy)
+
+        # Initialize calib repo
+        # Making the directory appears to be both necessary and sufficient
+        os.mkdir(self._calibRepo)
+
+    def _rawButler(self):
+        """Return multiple ways of querying calibration repositories.
+
+        Returns
+        -------
+        butler : `lsst.daf.persistence.Butler`
+            A butler that should all be capable of finding ingested science data.
+        """
+        return dafPersist.Butler(inputs={'root': self._repo, 'mode': 'r'})
+
+    def _calibButler(self):
+        """Return multiple ways of querying calibration repositories.
+
+        Returns
+        -------
+        butlers : `lsst.daf.persistence.Butler`
+            A butler that should all be capable of finding ingested calibration data.
+        """
+        return dafPersist.Butler(inputs={
+            'root': self._repo,
+            'mode': 'r',
+            'mapperArgs': {'calibRoot': self._calibRepo}})
+
+    def tearDown(self):
+        shutil.rmtree(self._repo, ignore_errors=True)
+
+    def testDataIngest(self):
+        """Test that ingesting a science image adds it to a repository.
+        """
+        files = [os.path.join(IngestionTestSuite.testData,
+                              'rawData',
+                              '2013-09-01',
+                              'z',
+                              'decam0229388.fits.fz')]
+        ingestion._doIngest(self._repo, IngestionTestSuite.testApVerifyData, files)
+
+        butler = self._rawButler()
+        self.assertTrue(butler.datasetExists('raw', dataId=IngestionTestSuite.rawDataId))
+
+    def testCalibIngest(self):
+        """Test that ingesting calibrations adds them to a repository.
+        """
+        files = [os.path.join(IngestionTestSuite.testData,
+                              'rawData',
+                              'cpCalib',
+                              'masterCal',
+                              calibFile) for calibFile in
+                 [os.path.join('fci.fits'),
+                  os.path.join('zci.fits')]
+                 ]
```

Comment:
This definition of files is difficult to make sense of over several lines. Consider defining testDataDir first and then grabbing the two files.